routes/products.js

```
@@ -59,10 +59,16 @@ router.get('/products/detail', function(req, res, next) {
 
 
 
+router.get('/products/search', function(req, res, next) {
+
+    check_logged(req, res);
 router.get('/products/search', function(req, res, next) {
 
     check_logged(req, res);
 
+    const url_params = url.parse(req.url, true).query;
+    const query = url_params.q;
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)